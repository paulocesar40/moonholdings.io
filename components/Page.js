import React, { Component } from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';

import Meta from '../components/Meta';
import { page } from '../styles';
import './global.scss';

const theme = {
  red: '#FF0000',
  black: '#393939',
  grey: '#3A3A3A',
  lightgrey: '#E1E1E1',
  offWhite: '#EDEDED',
  apricot: '#FEBE7E',
  margin: 0,
  padding: 0,
};

const { StyledPage, Inner } = page;

injectGlobal`
  html {
    height: 100%;
    box-sizing: border-box;
    font-size: 10px;
  }

  *, *:before, *:after  {
    box-sizing: inherit;
  }

  body {
    display: flex;
    overflow-y: auto;
    margin: 0;
    padding: 0;
    font-size: 1.5rem;
    line-height: 2;
    font-family: 'Abel', sans-serif;
  }

  a {
    text-decoration: none;
    color: ${theme.offWhite};
  }
`;

export default class Page extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <StyledPage className="app-bg">
          <Meta />
          <Inner>{this.props.children}</Inner>
        </StyledPage>
      </ThemeProvider>
    )
  }
}
