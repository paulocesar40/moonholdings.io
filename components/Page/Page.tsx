import React, { Component } from 'react'
import { ThemeProvider, injectGlobal } from 'styled-components'

import Meta from '../Meta/Meta'
import { StyledPage, Inner } from '../../styles'
import '../global.scss'

const theme = {
  red: '#FF0000',
  black: '#393939',
  grey: '#3A3A3A',
  midGray: '#515151',
  lightgrey: '#E1E1E1',
  offWhite: '#EDEDED',
  apricot: '#FEBE7E',
  brightPurple: '#B987C0',
  lightPurple: '#8F6894',
  darkPurple: '#57385C',
  margin: 0,
  padding: 0,
  transitionAll: 'all .2s ease-in-out',
};

injectGlobal`
  html {
    width: 100%;
    height: 100%;
    box-sizing: border-box;
    font-size: 16px;
  }

  *, *:before, *:after  {
    box-sizing: inherit;
  }

  body {
    margin: 0;
    padding: 0;
    font-family: 'Abel', sans-serif;
  }

  a {
    text-decoration: none;
    color: ${theme.offWhite};
  }
`;

export default class Page extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <StyledPage>
          <Meta />
          <Inner>{this.props.children}</Inner>
        </StyledPage>
      </ThemeProvider>
    )
  }
}
