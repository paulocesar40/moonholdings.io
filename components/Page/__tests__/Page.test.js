import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Page from '../Page'

describe('<Page /> component', () => {
  describe('rendering', () => {
    const wrapper = shallow(<Page />);
    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
    });
  });
});
