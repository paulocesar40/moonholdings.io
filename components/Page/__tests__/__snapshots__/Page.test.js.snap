// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> component rendering should render a component matching the snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "apricot": "#FEBE7E",
      "black": "#393939",
      "brightPurple": "#B987C0",
      "darkPurple": "#57385C",
      "grey": "#7E7E7E",
      "lightGrey": "#DADADA",
      "lightPurple": "#8F6894",
      "margin": 0,
      "midGray": "#A4A4A4",
      "offWhite": "#EDEDED",
      "padding": 0,
      "red": "#FF0000",
      "transitionAll": "all .2s ease-in-out",
    }
  }
>
  <page__StyledPage>
    <_default />
    <page__Inner />
  </page__StyledPage>
</ThemeProvider>
`;
